tests/pos/i19266.scala

```diff
@@ -0,0 +1,9 @@
+object i19266:
+  def fn1(x: Int, y: String = "y")(z: Double) = Some(s"$x$y$z")
+  def fn2(x: Int)(y: String) = Some(s"$x$y")
+
+  def checkCompile =
+    fn1(1)
```

Comment:
Scala 2.13.12 without `-Xsource:3` raises an error

> Unapplied methods are only converted to functions when a function type is expected.
> You can make this conversion explicit by writing `fn2 _` or `fn2(_)(_)` instead of `fn2`.

Reply:
https://github.com/lampepfl/dotty/blob/1716bcd9dbefbef88def848c09768a698b6b9ed9/compiler/src/dotty/tools/dotc/typer/Typer.scala#L3645

```
   *  (4) Do the following to unapplied methods used as values:
   *  (4.1) If the method has only implicit parameters pass implicit arguments
   *  (4.2) otherwise, if `pt` is a function type and method is not a constructor,
   *        convert to function by eta-expansion,
   *  (4.3) otherwise, if the method is nullary with a result type compatible to `pt`
   *        and it is not a constructor, apply it to ()
   *  otherwise issue an error
```

Reply:
https://github.com/lampepfl/dotty/blob/main/docs/_spec/06-expressions.md


> ### Method Conversions
> 
> The following four implicit conversions can be applied to methods which are not applied to some argument list.
> 
> ###### Evaluation
> A parameterless method ´m´ of type `=> ´T´` is always converted to type ´T´ by evaluating the expression to which ´m´ is bound.
> 
> ###### Implicit Application
> If the method takes only implicit parameters, implicit arguments are passed following the rules [here](07-implicits.html#implicit-parameters).
> 
> ###### Eta Expansion
> Otherwise, if the method is not a constructor, and the expected type ´\mathit{pt}´ is a function type, or, for methods of non-zero arity, a type [sam-convertible](#sam-conversion) to a function type, ´(\mathit{Ts}') \Rightarrow T'´, [eta-expansion](#eta-expansion-section) is performed on the expression ´e´.
> 
> (The exception for zero-arity methods is to avoid surprises due to unexpected sam conversion.)
> 
> ###### Empty Application
> Otherwise, if ´e´ has method type ´()T´, it is implicitly applied to the empty argument list, yielding ´e()´.
>

Reply:
> In the following example m can be partially applied to the first two parameters. __Assigning m to f1 will automatically eta-expand__
>
> ```scala
> def m(x: Boolean, y: String)(z: Int): List[Int]
> val f1 = m
> val f2 = m(true, "abc")
> ```


 https://dotty.epfl.ch/docs/reference/changed-features/eta-expansion-spec.html#automatic-eta-expansion-and-partial-application-1

Reply:
Note for myself.
https://github.com/lampepfl/dotty/blob/1716bcd9dbefbef88def848c09768a698b6b9ed9/compiler/src/dotty/tools/dotc/typer/Typer.scala#L4411